### DISCUSSION TITLE
Issues with yolov8 on raspberry pi 4

### DISCUSSION THREAD

**Discussion by scopeturned** [ultralytics/ultralytics#15699] [Q&A] [answered]
I have tried running yolov8 on my raspberry pi 4 after installing ultralytics and picamera2 on a headless version of raspbian but when i try to run `from ultralytics import YOLO` it gives me the error **Illegal instruction**.

**Comment by glenn-jocher** [comment]
@scopeturned please ensure your Raspberry Pi 4 is running the latest Raspberry Pi OS Bookworm (Debian 12) and that you have the latest version of the Ultralytics package installed. If the issue persists, try reinstalling the package using `pip install --force-reinstall ultralytics`. For more details, visit our [Raspberry Pi guide](https://docs.ultralytics.com/guides/raspberry-pi/).

**Comment by scopeturned** [comment]
All the packages were up to date and running on debian 12. Even if this is the case i still get the **Illegal instruction** error.

**Reply by glenn-jocher** [reply, marked as answer]
@scopeturned thank you for reporting this issue. The **Illegal instruction** error often indicates compatibility problems with the CPU architecture. Please ensure you are using the latest version of the Ultralytics package. If the issue persists, try running the code in a Docker container optimized for Raspberry Pi as outlined in our [Raspberry Pi guide](https://docs.ultralytics.com/guides/raspberry-pi/).

**Reply by scopeturned** [reply]
Thank you for the help! I installed it using docker and now it works, although i'm still working to get the picamera2 library to import properly into docker.

**Reply by glenn-jocher** [reply]
@scopeturned glad to hear it's working with Docker! For the picamera2 library, ensure it's installed within the Docker container and that the container has access to the camera hardware. If you encounter specific issues, feel free to share them here.
